Sources/Search/OASearchUICore.mm

```diff
@@ -201,13 +202,29 @@ -(NSComparisonResult) compare:(OASearchResult *)o1 o2:(OASearchResult *)o2 compa
             }
             else if (am1 && am2)
             {
+                // here 2 points are amenity
                 const auto& a1 = std::dynamic_pointer_cast<const OsmAnd::Amenity>(o1.amenity);
                 const auto& a2 = std::dynamic_pointer_cast<const OsmAnd::Amenity>(o2.amenity);
-                NSComparisonResult cmp = (NSComparisonResult)OsmAnd::ICU::ccompare(a1->type, a2->type);
+                
+                QString type1 = a1->type;
```

Comment:
Doesn't seem to reflect the latest changes in Java (rev. 3e9ef3914b3b3962d00e2803bdd2ad6411cb6523)